.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,90 @@
+---
+# Full PR suite.  PRs need to pass this pipeline in order to be moved to the
+# merge queue, where they'll use the next rule as the branch-protection rule
+# for the final merge to the base branch.
+name: Pull Requests
+on:
+  pull_request:
+    branches: [main, "stable/*"]
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  lint-docs:
+    if: github.repository_owner == 'Qiskit'
+    name: Docs and Lint
+    uses: ./.github/workflows/lint_docs.yml
+  main-tests-ubuntu-3-9:
+    name: Main Tests
+    uses: ./.github/workflows/test-linux.yml
+    if: github.repository_owner == 'Qiskit'
+    with:
+      python-version: "3.9"
+      install-optionals: true
+      runner: ubuntu-latest
+      use-msrv: true
+  main-tests-rust:
+    name: Extra Tests
```

Comment:
What does extra tests mean? These seem like all part of the main workflow. Do you mean non-python unit test jobs? I think that should be the distinction then. But I think instead of just saying extra tests here "Rust unit tests" and "Image comparison tests" for the names here would be better.

Reply:
Changed to "Rust Unit Tests" and "Image Comparison Tests" in b7f717b